Utils/Dataflow/017_adjustMetadata/adjustMetadata.py

```diff
@@ -153,6 +153,76 @@ def input_events(data):
     return result
 
 
+def input_events_v2(data):
+    """ Calculate derived value 'input_events_v2'.
+
+    For the first tasks in a chain:
+     * total_req_events (if defined);
+     * else: primary_input_events (if defined);
+     * else: requested_events.
+
+    For other tasks:
+     * parent_total_events (if parent is not derivation);
+     * else: requested_events.
+
+    For details please contact Mikhail Borodin <Mikhail.Borodin@cern.ch>.
+
+    :param data: task metadata
+    :type data: dict
+
+    :return: derived value 'input_events_v2'
+             (None if input data do not provide enough information)
+    :rtype: int, NoneType
+    """
+    result = None
+
+    # First task in chain
+    if data['taskid'] == data['chain_id']:
```

Comment:
1. If something goes wrong (as in “there’s no `taskid` or `chain_id`), then we can not calculate values we need properly, and, in fact, there’s something completely wring about given message. If it’s a one-time accident, we fail now and try again later when the process restarted. If we keep on failing at this record than it’s not an accident and we need to take a closer look at the data and decide how to handle it. When it’s no longer a hypothetical “what if”.

2. This”special treatment” means “we need to know if this task is a first in chain or not. If we don’t know for sure, but set `chain_id` to `taskid`, here we’ll think that it _is_ first, although it might as well be the opposite. I believe it would be incorrect.